test/test_stringUtil.cpp

```diff
@@ -0,0 +1,108 @@
+#include <catch2/catch_test_macros.hpp>
+
+#include "renderlib/StringUtil.h"
+
+TEST_CASE("StringUtil", "[StringUtil]")
+{
+  SECTION("trim")
+  {
+    REQUIRE(trim("  hello  ") == "hello");
+    REQUIRE(trim("  hello") == "hello");
+    REQUIRE(trim("hello  ") == "hello");
+    REQUIRE(trim("hello") == "hello");
+    REQUIRE(trim("  ") == "");
+    REQUIRE(trim("") == "");
+    REQUIRE(trim("  \t\n\r  ") == "");
+    REQUIRE(trim("foobarfoobar", "foo") == "barfoobar");
+    REQUIRE(trim("foobarfoo", "foo") == "bar");
+    REQUIRE(trim("foobaroof", "foo") == "bar");
+  }
+  SECTION("startsWith")
+  {
+    REQUIRE(startsWith("hello", "he") == true);
+    REQUIRE(startsWith("hello", "hello") == true);
+    REQUIRE(startsWith("hello", "hello ") == false);
+    REQUIRE(startsWith("hello", "hellox") == false);
+    REQUIRE(startsWith("hello", "xhello") == false);
+    REQUIRE(startsWith("hello", "x") == false);
+    REQUIRE(startsWith("hello", "") == true);
+    REQUIRE(startsWith("", "hello") == false);
+    REQUIRE(startsWith("", "") == true);
+  }
+  SECTION("endsWith")
+  {
+    REQUIRE(endsWith("hello", "lo") == true);
+    REQUIRE(endsWith("hello", "hello") == true);
+    REQUIRE(endsWith("hello", " hello") == false);
+    REQUIRE(endsWith("hello", "xhello") == false);
+    REQUIRE(endsWith("hello", "x") == false);
+    REQUIRE(endsWith("hello", "") == true);
+    REQUIRE(endsWith("", "hello") == false);
+    REQUIRE(endsWith("", "") == true);
+  }
+  SECTION("split")
+  {
+    std::vector<std::string> elems;
+    split("hello world", ' ', elems);
+    REQUIRE(elems.size() == 2);
+    REQUIRE(elems[0] == "hello");
+    REQUIRE(elems[1] == "world");
+
+    elems.clear();
+    split("hello world", 'x', elems);
+    REQUIRE(elems.size() == 1);
+    REQUIRE(elems[0] == "hello world");
+
+    elems.clear();
+    split("hello world", 'o', elems);
+    REQUIRE(elems.size() == 3);
+    REQUIRE(elems[0] == "hell");
+    REQUIRE(elems[1] == " w");
+    REQUIRE(elems[2] == "rld");
+
+    elems.clear();
+    split("name=value", '=', elems);
+    REQUIRE(elems.size() == 2);
+    REQUIRE(elems[0] == "name");
+    REQUIRE(elems[1] == "value");
+
+    elems.clear();
+    split("name=value\nname2=value2\n", '\n', elems);
+    REQUIRE(elems.size() == 2);
+    REQUIRE(elems[0] == "name=value");
+    REQUIRE(elems[1] == "name2=value2");
+  }
```

Comment:
Hmm, this is interesting to me. I think in JS the expected behavior is `["name=value", "name2=value2", ""]` where the ending `\n` would cause an empty string at the end? Any reason to just outright discard it?

Reply:
Especially since it looks like you're adding a check for empty strings in `splitToNameValuePairs`...

Reply:
I added an extra test that shows empty string.  This is just how my implementation works.  It's using std::getline with the specified delimiter.  When the delimiter is at the end of the string, you don't get any tokens after that.